sympy/solvers/solvers.py

```diff
@@ -2653,22 +2655,53 @@ def _tsolve(eq, sym, **flags):
                     return _solve(lhs - rhs, sym, **flags)
             elif sym not in lhs.exp.free_symbols:
                 return _solve(lhs.base - rhs**(1/lhs.exp), sym, **flags)
-            elif not rhs and sym in lhs.exp.free_symbols:
-                # f(x)**g(x) only has solutions where f(x) == 0 and g(x) != 0 at
-                # the same place
-                sol_base = _solve(lhs.base, sym, **flags)
-                if not sol_base:
-                    return sol_base  # no solutions to remove so return now
-                return list(ordered(set(sol_base) - set(
-                    _solve(lhs.exp, sym, **flags))))
-            elif (rhs is not S.Zero and
-                        lhs.base.is_positive and
-                        lhs.exp.is_real):
-                return _solve(lhs.exp*log(lhs.base) - log(rhs), sym, **flags)
-            elif lhs.base == 0 and rhs == 1:
-                return _solve(lhs.exp, sym, **flags)
             else:
-                raise NotImplementedError
+                # sym in lhs.exp.free_symbols
+                if not rhs:
+                    # f(x)**g(x) only has solutions where f(x) == 0 and g(x) != 0 at
+                    # the same place
+                    sol_base = _solve(lhs.base, sym, **flags)
+                    if not sol_base:
+                        return sol_base  # no solutions to remove so return now
+                    return list(ordered(set(sol_base) - set(
+                    _solve(lhs.exp, sym, **flags))))
+                elif lhs.base.is_positive and lhs.exp.is_real:
+                    return _solve(lhs.exp*log(lhs.base) - log(rhs), sym, **flags)
+                elif rhs.is_Rational:
+                    if  lhs.base == 0:
+                        return _solve(lhs.exp, sym, **flags)
+                    elif lhs.base.is_Number and lhs.base>0:
+                        rewrite = lhs.rewrite(exp)
+                        if rewrite != lhs:
+                            return _solve(rewrite - rhs, sym, **flags)
+                    else:
+                        sol = []
+                        temp_num=[]
+                        temp_pow=[]
+                        b_pos=[]
+                        b_negi=[]
+                        e0=[]
+                        for i in divisors(rhs.p):
+                            b=0
+                            t=True
+                            if i!=1:
+                                b, t = integer_log(rhs.p,i)
+                            if t:
+                                for s in divisors(rhs.q):
+                                    if s**b== rhs.q:
+                                        temp_num.append(Rational(i,s))
+                                        temp_pow.append(b)
+                        for i in temp_num:
```

Comment:
It doesn't look like there is a need to store `temp_num`: just move the extend-ing lines below to where you compute the temps above.